File: MyFirstAPP/爱奇艺/基本类库/IqiyiNavigationController.m
//
//  IqiyiNavigationController.m
//  MyFirstAPP
//
//  Created by 薛超 on 16/9/2.
//  Copyright © 2016年 薛超. All rights reserved.
//

#import "IqiyiNavigationController.h"

@implementation IqiyiNavigationController
- (void)viewDidLoad {
    [super viewDidLoad];
    self.view.backgroundColor = [UIColor whiteColor];
    self.navigationBar.translucent = NO;//不是半透明
    [self.navigationBar setTitleTextAttributes:[NSDictionary dictionaryWithObjectsAndKeys:[UIColor whiteColor], NSForegroundColorAttributeName, Font(16), NSFontAttributeName, nil]];
    self.navigationBar.barTintColor = IqiyinavigationBarColor;
    if( ([[[UIDevice currentDevice] systemVersion] doubleValue]>=7.0)){
        self.edgesForExtendedLayout = UIRectEdgeNone;//视图控制器，四条边不指定
        self.extendedLayoutIncludesOpaqueBars = NO;//不透明的操作栏
        self.modalPresentationCapturesStatusBarAppearance = NO;
        [[UINavigationBar appearance] setBackgroundImage:[UIImage imageNamed:@""] forBarPosition:UIBarPositionTop barMetrics:UIBarMetricsDefault];
    }else{
        [self.navigationBar setBackgroundImage:[UIImage imageNamed:@""] forBarMetrics:UIBarMetricsDefault];
    }
}

@end

File: Freedom/爱奇艺/基本类库/IqiyiNavigationController.swift
//
//  IqiyiNavigationController.swift
//  Freedom
//
//  Created by htf on 2018/5/17.
//  Copyright © 2018年 薛超. All rights reserved.
//

import UIKit

class IqiyiNavigationController: XBaseNavigationViewController {
    view.backgroundColor = UIColor.white
    navigationBar.isTranslucent = false
    //不是半透明
    navigationBar.titleTextAttributes = [
    NSForegroundColorAttributeName : UIColor.white,
    NSFontAttributeName : Font(16)
    ]
    navigationBar.barTintColor = RGBCOLOR(47, 47, 47)
    if (Double(UIDevice.current.systemVersion) ?? 0.0 >= 7.0) {
    edgesForExtendedLayout = []
    //视图控制器，四条边不指定
    extendedLayoutIncludesOpaqueBars = false
    //不透明的操作栏
    modalPresentationCapturesStatusBarAppearance = false
    UINavigationBar.appearance().setBackgroundImage(UIImage(named: ""), for: .top, barMetrics: .default)
    } else {
    navigationBar.setBackgroundImage(UIImage(named: ""), for: .default)
    }

}
